### PR TITLE
Introduce documentation site

### PR DESCRIPTION
**Pull request by legrego**
Adds a documentation site powered by `material-mkdocs`.

Preview docs at https://legrego.github.io/homeassistant-elasticsearch/

TODO:
- Cleanup `README` and refer to docs site for more information

### CODE REVIEW

File: docs/configure.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          "hass_writer": {
+            "cluster": [
+              "manage_index_templates",
+              "manage_ilm",
+              "monitor"
+            ],
+            "indices": [
+              {
+                "names": [
+                  "metrics-homeassistant*"
```
Comment (strawgate) [inline docs/configure.md:35]: should be `metrics-homeassistant.*`

File: docs/configure.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        "hass_writer": {
+          "cluster": [
+            "manage_index_templates",
+            "manage_ilm",
+            "monitor"
+          ],
+          "indices": [
+            {
+              "names": [
+                "metrics-homeassistant*"
```
Comment (strawgate) [inline docs/configure.md:69]: metrics-homeassistant.*

File: docs/configure.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+### Entity selection
+You can choose to include/exclude specific entities and/or entire domains. This allows you to publish only the entities you are interested in to Elasticsearch.
+By default, all entities and domains are included.
+You can combine inclusion and exclusion filters to fine-tune the entities you want to publish.
+
+The following flowchart describes the logic used to determine if an entity is published:
+
+```mermaid
+graph LR
```
Comment (strawgate) [inline docs/configure.md:131]: see https://github.com/legrego/homeassistant-elasticsearch/issues/273 -- I think we should change this logic

File: docs/ingest_advanced.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        '
+    ```
+
+    1. Replace `https://localhost:9200` with the URL of your Elasticsearch instance
+    2. Replace `elastic` with your Elasticsearch username
+    3. Replace `changeme` with your Elasticsearch password
+    4. Add your ingest pipeline processors to the `processors` array
+
+Component template changes apply when the datastream performs a rollover so the first time you modify the template you may need to manually initiate ILM rollover to start applying the pipeline.
```
Comment (strawgate) [inline docs/ingest_advanced.md:73]: initiate index/datastream rollover instead of ILM rollover

Comment (legrego) [reply docs/ingest_advanced.md]: Done in https://github.com/legrego/homeassistant-elasticsearch/pull/268/commits/861ac46421ca6043a2f1c73c77669471885039a5

File: docs/configure.md (hunk 2 above)
Comment (legrego) [reply docs/configure.md]: Done in https://github.com/legrego/homeassistant-elasticsearch/commit/861ac46421ca6043a2f1c73c77669471885039a5

File: docs/configure.md (hunk 1 above)
Comment (legrego) [reply docs/configure.md]: Done in https://github.com/legrego/homeassistant-elasticsearch/commit/861ac46421ca6043a2f1c73c77669471885039a5

File: docs/configure.md (hunk 3 above)
Comment (legrego) [reply docs/configure.md:131]: Let's continue the discussion in #273. I'll adjust this diagram once we settle on an implementation.

**Pull request merged**